mypy/checker.py

```diff
@@ -1308,7 +1308,10 @@ def check_override(self, override: FunctionLike, original: FunctionLike,
         """
         # Use boolean variable to clarify code.
         fail = False
-        if not is_subtype(override, original, ignore_pos_arg_names=True):
+        # __call__ is special among dunders, because arguments can be passed
+        # as keyword args (unlike other dunders).
+        ignore_names = name != '__call__'
+        if not is_subtype(override, original, ignore_pos_arg_names=ignore_names):
```

Comment:
Hmm, this could break some existing code which assumes that `__call__` won't be called with keyword arguments. Maybe there is a way to restrict this to only protocol classes, where `__call__` is special among dunder methods?